src/bin/client.rs

```diff
@@ -2,88 +2,89 @@ extern crate piston;
 extern crate graphics;
 extern crate opengl_graphics;
 extern crate rand;
-#[cfg(feature = "include_sdl2")]
-extern crate sdl2_window;
-#[cfg(feature = "include_glfw")]
-extern crate glfw_window;
-#[cfg(feature = "include_glutin")]
-extern crate glutin_window;
+extern crate rpsrtsrs;
+#[cfg(feature = "include_sdl2")] extern crate sdl2_window;
+#[cfg(feature = "include_glfw")] extern crate glfw_window;
+#[cfg(feature = "include_glutin")] extern crate glutin_window;
 
+
+use std::f64::consts::PI;
 use piston::window::WindowSettings;
 use opengl_graphics::{ GlGraphics, OpenGL };
 use piston::input::*;
 use piston::event_loop::*;
-#[cfg(feature = "include_sdl2")]
-use sdl2_window::Sdl2Window as Window;
-#[cfg(feature = "include_glfw")]
-use glfw_window::GlfwWindow as Window;
-#[cfg(feature = "include_glutin")]
-use glutin_window::GlutinWindow as Window;
-
-pub struct Square {
-    rotation: f64,   // Rotation for the square.
-    position: [f64; 2],
-    target: [f64; 2],
-    selected: bool,
-}
-
-impl Square {
-    fn new(position: [f64;2]) -> Square {
-        println!("Create square at {:?}", position);
-        Square {
-            rotation: 0.0,
-            position: position,
-            target: position,
-            selected: false
-        }
-    }
-}
+#[cfg(feature = "include_sdl2")] use sdl2_window::Sdl2Window as Window;
+#[cfg(feature = "include_glfw")] use glfw_window::GlfwWindow as Window;
+#[cfg(feature = "include_glutin")] use glutin_window::GlutinWindow as Window;
+use rpsrtsrs::shapes::Unit;
 
 pub struct App {
     gl: GlGraphics, // OpenGL drawing backend.
-    squares: Vec<Square>,
+    units: Vec<Unit>,
 }
 
 impl App {
-    fn select(&mut self, position: &[f64;2]) {
-        for s in &mut self.squares {
-            s.selected = position[0]< s.position[0]+25.0 &&
-                position[0]> s.position[0]-25.0 &&
-                position[1]< s.position[1]+25.0 &&
-                position[1]> s.position[1]-25.0;
+    fn select(&mut self, position: [f64;2]) {
```

Comment:
See http://words.steveklabnik.com/pointers-in-rust-a-guide. The article is quite old and outdated, but contains some interesting parts about using pass-by-value instead of pass-by-reference:

> But won’t this be inefficent? Well, that’s a complicated question, but it’s important to know that Rust, like C and C++, store aggregate data types ‘unboxed,’ whereas languages like Java and Ruby store these types as 'boxed.’ For smaller structs, this way will be more efficient. For larger ones, it may be less so. But don’t reach for that pointer until you must! Make sure that the struct is large enough by performing some tests before you add in the complexity of pointers.

I'm not sure how much it also applies to borrowed pointers, because they're checked at compile time. But it does make the signature easier to read. And we definitely don't need to optimize the performance of accessing `[f64;2]`.